### ISSUE TITLE
cloud-config write_files doesn't appear to support the encoding attribute

### ISSUE THREAD

**Issue by antonosmond** [rancher/os#1707]
**RancherOS Version:** v0.9.0

**Where are you running RancherOS?** AWS

I'm trying to run rancher server on RancherOS and my cloud-config contains the following write-files block:

```
#cloud-config
write_files:
  - path: /var/lib/rancher/etc/ssl/ca.crt
    permissions: "0644"
    owner: root
    encoding: b64
    content: YmxhaAo=
```

The file is never created and if I SSH onto the AWS instance and run cloud-init-execute manually, I get an error stating `Unable to write file with encoding b64`.

According to the cloud-init docs [here](https://cloudinit.readthedocs.io/en/latest/topics/examples.html#writing-out-arbitrary-files), write-files should support encoding and it's a feature I've used in other linux distros with cloud-init. Is this something that the RancherOS cloud-init process should support and if not is it something that will be supported in a future release?

Thanks!

Anton

**Comment by quantverse** [comment]
I can confirm this used to work before (in Rancher 0.5 era). Now upgraded to 1.0.0 and this does not work anymore. I need to redesign my cloud-configs now.

**Comment by mvisonneau** [comment]
Is this related to : https://github.com/rancher/os/issues/2284 ? 🤔

**Comment by Jason-ZW** [comment]
@mvisonneau Yes. This issue should be fixed at 1.4.0-rc1

**Comment by Jason-ZW** [comment]
See #2284. Close this issue.
